internal/testrunner/runners/system/tester.go

```diff
@@ -1978,6 +1985,248 @@ func (r *tester) previewTransform(ctx context.Context, transformId string) ([]co
+		Details         string    `json:"details"`
+		Count           int       `json:"count"`
+		FirstOccurrence time.Time `json:"first_occurrence"`
+	} `json:"issues"`
+}
+
+func (r *tester) getTransformStats(ctx context.Context, transformId string) (*transformStats, error) {
+	resp, err := r.esAPI.TransformGetTransformStats(transformId,
+		r.esAPI.TransformGetTransformStats.WithContext(ctx),
+		r.esAPI.TransformGetTransformStats.WithAllowNoMatch(false),
+	)
+
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.IsError() {
+		return nil, fmt.Errorf("failed to get transform stats for %q: %s", transformId, resp.String())
+	}
+
+	var response struct {
+		Transforms []transformStats `json:"transforms"`
+	}
+	err = json.NewDecoder(resp.Body).Decode(&response)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode response: %w", err)
+	}
+	if len(response.Transforms) != 1 {
+		return nil, fmt.Errorf("stats for %d transforms received when requesting only for %s", len(response.Transforms), transformId)
+	}
+	return &response.Transforms[0], nil
+}
+
+func (r *tester) checkTransformAuditMessages(ctx context.Context, transformId string) error {
+	// XXX: This is an internal API, are these audit messages available somewhere else?
+	const internalTransformsPath = "/internal/transform/transforms"
+	messagesPath := path.Join(internalTransformsPath, transformId, "messages")
+	query := "?sortField=timestamp&sortDirection=desc" // Required
+	statusCode, body, err := r.kibanaClient.SendRequest(ctx, http.MethodGet, messagesPath+query, nil)
+	if err != nil {
+		return fmt.Errorf("could not get transform audit messages: %w", err)
+	}
+	if statusCode >= 400 {
+		return fmt.Errorf("could not get transform audit messages: status code %d, body: %s", statusCode, body)
+	}
+
+	var resp struct {
+		Messages []struct {
+			TransformID string `json:"transform_id"`
+			Message     string `json:"message"`
+			Level       string `json:"level"`
+			Timestamp   int    `json:"timestamp"`
+			NodeName    string `json:"node_name"`
+		} `json:"messages"`
+	}
+	err = json.Unmarshal(body, &resp)
+	if err != nil {
+		return fmt.Errorf("could not decode response: %w", err)
+	}
+
+	for _, message := range resp.Messages {
+		if message.Level == "error" {
+			return fmt.Errorf("failure found in transform: %s", message.Message)
+		}
+	}
+	return nil
+}
+
+// checkRunningTransformHealth checks the following for a given transform:
+// - That it is started.
+// - That it can execute at least once during the check.
+// - That it hasn't generated any error message.
+// - That it is healthy after executing at least once.
+func (r *tester) checkRunningTransformHealth(ctx context.Context, transformId string) error {
+	const (
+		period  = 1 * time.Second
+		timeout = 60 * time.Second
+	)
+	lastSearchTime := 0
+	last := -1
+	running := false
+
+	// Reset transform to clean any previous state.
+	/* XXX: It fails to create the index after reset :?
```

Comment:
This is failing now because it tries to create indexes with a name that is matched by a data stream template, and transforms need to use indexes, not data streams.

:warning: This may be an actual issue because this means that the reset API breaks these transforms. We may need to review how transform indexes are named in packages to avoid these conflicts.